### DISCUSSION TITLE
Tunnel without python

### DISCUSSION THREAD

**Discussion by whoim2** [svpcom/wfb-ng#295] [General]
Hello! Im try use wfb-ng at embedded linux devices, air and ground. Video and telemetry work nice, but im not have tunnel from ground to air. Devices have tun driver. What i may create tun interfaces and link it to wfb processes?

**Comment by svpcom** [comment]
Tunnel is implemented in python part of code. To setup tunnel without python you need to implement it in C or use any tunnel with UDP encapsulation (for example vtund) on **both** sides

**Reply by whoim2** [reply]
vtund present in my devices, but what i may link this separetad to rx/tx processed?
I need 2 pairs, tx>rx and rx>tx. Its 2 udp ports on each side.

**Reply by whoim2** [reply]
![Screenshot_10](https://user-images.githubusercontent.com/13419599/226569989-95410fbf-b372-494b-8a38-ec983d87e8ae.png)
im try as this picture, thanks

**Reply by svpcom** [reply]
Yes, you need to patch vtund to use separate tx/rx udp ports

**Reply by whoim2** [reply]
> Yes, you need to patch vtund to use separate tx/rx udp ports

Oh.. its not really for me. Thanks.
